sdk/storage/Azure.Storage.Blobs/migrationGuides/AzureStorageNetMigrationV12.md

```diff
@@ -0,0 +1,272 @@
+
+Legacy (v11)
+```c
+TODO
+```
+
+v12
+```c
+// This code snippet creates a service level SAS that only allows reading
+// from service level APIs
+AccountSasBuilder sas = new AccountSasBuilder
+{
+    // Allow access to blobs
+    Services = AccountSasServices.Blobs,
+
+    // Allow access to the service level APIs
+    ResourceTypes = AccountSasResourceTypes.Service,
+
+    // Access expires in 1 hour!
+    ExpiresOn = DateTimeOffset.UtcNow.AddHours(1)
+};
+// Allow read access
+sas.SetPermissions(AccountSasPermissions.Read);
+
+// Create a SharedKeyCredential that we can use to sign the SAS token
+StorageSharedKeyCredential credential = new StorageSharedKeyCredential(StorageAccountName, StorageAccountKey);
+
+// Build a SAS URI
+UriBuilder sasUri = new UriBuilder(StorageAccountBlobUri);
+sasUri.Query = sas.ToSasQueryParameters(credential).ToString();
+
+// Create a client that can authenticate with the SAS URI
+BlobServiceClient service = new BlobServiceClient(sasUri.Uri);
+
+// Make a service request to verify we've successfully authenticated
+await service.GetPropertiesAsync();
+```
+
+#### Connection string
+
+The following code assumes you have acquired your connection string (you can do so from the Access Keys tab under Settings in your Portal Storage Account blade). It is recommended to store it in an environment variable.
+
+[//]: # (Parsing the connection string in v11 vs v12.....)
+
+Legacy (v11)
+```c
+string connectionString = Environment.GetEnvironmentVariable("AZURE_STORAGE_CONNECTION_STRING");
+
+// Check whether the connection string can be parsed.
+CloudStorageAccount storageAccount;
+if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
+{
+    // If the connection string is valid, proceed with operations against Blob
+    // storage here.
+}
+else
+{
+    // Otherwise, user needs to define the environment variable.
+}
+```
+
+v12
+```c
+string connectionString = Environment.GetEnvironmentVariable("AZURE_STORAGE_CONNECTION_STRING");
+
+// Create a client that can authenticate with a connection string
+BlobServiceClient service = new BlobServiceClient(connectionString);
+
+// Make a service request to verify we've successfully authenticated
+await service.GetPropertiesAsync();
+```
+
+### Shared Access Policies
+
+TODO
+
+### Client hierarchy
+
+In the interest of simplifying the API surface we've made a three top level clients that can be used to interact with a majority of your resources: `BlobServiceClient`, `BlobContainerClient`, and `BlobClient`.
+
+[//]: # (Blob Metadata, properties, and attributes...)
```

Comment:
Is this going to pull in some external articles? Given that changes associated with metadata/properties/attributes are breaking changes, this is the meat of the upgrade document.